### PR TITLE
refactor e2e for speed and clarity, add helpers

### PR DESCRIPTION
**Pull request by joelanford**
# Description

This PR improves the install e2e test by:
- Using a smaller and controlled-by-us catalog image (though it is still hosted/pulled from quay). This reduces e2e time from ~9m to ~5m.
- Adding helpers to make it easier to check conditions
- `Eventually` blocks now fail early if unexpected failures occur, which makes e2e runs fail faster rather than being required to wait the full `Eventually` timeout.

This should mostly eliminate the flakes related to catalog processing mentioned in #215.

## Reviewer Checklist

- [ ] API Go Documentation
- [x] Tests: Unit Tests (and E2E Tests, if appropriate)
- [x] Comprehensive Commit Messages
- [ ] Links to related GitHub Issue(s)

### CODE REVIEW

File: test/e2e/install_test.go
Comment (joelanford) [inline test/e2e/install_test.go:132]: An earlier iteration of this section of code called a separate function I had written that generically:
- converted to unstructured
- extracted "status.conditions" into `[]interface{}`
- marshaled that to JSON
- unmarshaled the JSON into []metav1.Condition

That code generically handled _anything_ that had `status.conditions` with type `[]metav1.Condition`, but it seemed a bit hacky/overkill. I'm happy to go back to that, or entertain other options as well.

File: test/e2e/install_test.go
Comment (everettraven) [inline test/e2e/install_test.go:129]: nit:
```suggestion
	case *catalogd.Catalog, *operatorv1alpha1.Operator:
		conditions = v.Status.Conditions
```

**Review by everettraven — APPROVED** [review]
super teeny non-blocking nit, otherwise looks good to me

**Comment by joelanford** [comment]
/hold
Until after #205 merges (and I'll rebase afterward)

File: test/e2e/install_test.go
Comment (everettraven) [reply test/e2e/install_test.go:129]: Oh that's interesting. TIL

**Pull request closed without merging**
